egg/zoo/compo_vs_generalization/curriculum_games.py

```diff
@@ -104,23 +116,25 @@ def forward(self, sender_input, labels, receiver_input=None, aux_input=None):
 
         idxs_to_reveal = idxs_to_reveal.to(device)
 
-        # mask indices with a redundant value
-        # e.g. if idxs_to_reveal = [[3, 2, 0, 1]] and n_revealed = [[2]]
-        # we define mask_idxs_to_reveal = [[1, 1, 0, 0]]
-        # and we mask idxs_to_reveal with its first value, so as to get
-        # idxs_to_reveal = [[3, 2, 3, 3]].
-        # this way, we only have 2 indices to reveal, but we keep
-        # a consistent shape within the batch.
-        mask_idxs_to_reveal = torch.arange(self.n_attributes).expand(
-            batch_size, self.n_attributes
-        ).to(device)
-        mask_idxs_to_reveal = (
-                mask_idxs_to_reveal < n_revealed.view(batch_size, -1)
-        ).long()
-        idxs_to_reveal = idxs_to_reveal * mask_idxs_to_reveal
-        idxs_to_reveal = idxs_to_reveal + idxs_to_reveal[:, 0].view(
-            batch_size, 1
-        ).expand(batch_size, self.n_attributes) * (1 - mask_idxs_to_reveal)
+        # only mask attributes when training
+        if not self.testing:
```

Comment:
I'm not sure if this should be handled this way: Ideally you'd like to retain the ability to test with masks as well (or test with a different number of masks).

Here's what I'd suggest:
1. Quick & dirty solution: have an optional argument: `mask_during_eval` that you can set to True or False
2. More fleshed out solution (maybe in another PR?): Don't change the behaviour here. Instead, create a dedicated callback that implements:
    - `on_validation_begin()`: change the parameters of the `game` object to what you want to evaluate on (number of masks, distribution, etc...)
    - `on_validation_end()`: reset the game parameters to what is used for training

Reply:
Good point, it would also give the possibility to do several different tests with a different number of masks. I will make another PR addressing this.